client/lists/infinite-scroll-list.tsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react'
+import styled from 'styled-components'
+import LoadingIndicator from '../progress/dots'
+
+const LoadingArea = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: center;
+  min-height: 56px;
+  padding: 16px 0;
+`
+
+interface InfiniteListProps {
+  /** Whether the functionality of loading more data at the beginning of the list is enabled. */
+  prevLoadingEnabled?: boolean
+  /** Whether the functionality of loading more data at the ending of the list is enabled. */
+  nextLoadingEnabled?: boolean
+  /** Whether we are currently loading more data at the beginning of the list. */
+  isLoadingPrev?: boolean
+  /** Whether we are currently loading more data at the ending of the list. */
+  isLoadingNext?: boolean
+  /** Whether this list has more data that could be loaded at the beginning of the list. */
+  hasMorePrevData?: boolean
```

Comment:
nit: maybe `hasPrevData`/`loadPrevData` and the same for `Next`? The 'more' makes these names pretty weird and I don't think adds much to the meaning